.github/workflows/test.yml

```diff
@@ -85,6 +85,19 @@ jobs:
           path: build
           if-no-files-found: error
 
+  build-e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup
+      - run: yarn prepare
+      - run: yarn build:e2e
+      - uses: actions/upload-artifact@v2
+        with:
+          name: build-e2e
+          path: build
+          if-no-files-found: error
+
   size-tests:
     needs: [build]
     runs-on: ubuntu-latest
```

Comment:
Can you move this to be in the same step as `build`, and rename `build` to `size-tests`? Then you won't need the upload action in it, and it should make it run faster overall.